stacks/minstack.js

```diff
@@ -0,0 +1,39 @@
+const Stack = require('./Stack');
+
+function MinStack(stackMaxSize){
+    this.stack = new Stack(stackMaxSize);
+    this.minStack = new Stack(stackMaxSize);
+}
+
+MinStack.prototype.push = function(num) {
+    var currentMinimum = this.minStack.peek();
+    if(this.minStack.count() === 0 || (!currentMinimum || currentMinimum > num) ) //First element
+    {
+        this.minStack.push(num);
+    }
+    else
+    {
+        this.minStack.push(currentMinimum);
+    }
+    this.stack.push(num);
+}
+
+MinStack.prototype.pop = function() {
+    this.minStack.pop();
+    this.stack.pop();
```

Comment:
this should be returning stack.pop right?

Reply:
here was not using the value of popped element. so did not return the value.

Reply:
a pop always returns value that got deleted

Reply:
added now